nvflare/app_opt/xgboost/histogram_based/executor.py

```diff
@@ -243,9 +243,10 @@ def train(self, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -
         self.world_size = world_size
 
         if self.use_gpus:
-            # mapping each rank to a GPU (can set to cuda:0 if simulating with only one gpu)
-            self.log_info(fl_ctx, f"Training with GPU {self.rank}")
-            self.xgb_params["device"] = f"cuda:{self.rank}"
+            # mapping each rank to a GPU if not set (can set to cuda:0 if simulating with only one gpu)
+            device = self.xgb_params.get("device", f"cuda:{self.rank}")
```

Comment:
I'm not sure we should assume the default device id to be client rank. What if there are many clients?
Perhaps it's safer to assume device 0.

Reply:
good suggestion, make the suggested changes in https://github.com/NVIDIA/NVFlare/pull/2787
closing this pr